operator/internal/securedcluster/values/translation/translation.go

```diff
@@ -334,6 +334,37 @@ func (t Translator) getCollectorContainerValues(collectorContainerSpec *platform
 	}
 
 	cv.AddChild(translation.ResourcesKey, translation.GetResources(collectorContainerSpec.Resources))
+	runtimeConfig := collectorContainerSpec.RuntimeConfig
+	if runtimeConfig != nil {
+		cv.AddChild("runtimeConfig", t.getRuntimeConfig(runtimeConfig))
+	}
+
+	return &cv
+}
+
+func (t Translator) getRuntimeConfig(runtimeConfig *platform.CollectorRuntimeConfig) *translation.ValuesBuilder {
```

Comment:
This code means that anytime a new field is added to collector runtime config, it will need to be added here. That does not seem like the best solution.

Reply:
This code translates between:
- user-centric configuration surface, where certain conventions and consistency with the rest of the CRD apply, and
- the collector-centric proto definition where such concerns might be irrelevant.

So I don't see a way around this. That is, unless the collector team considers these conventions from the very beginning when designing each new user-controllable feature..

There is the unfortunate helm layer in between as well, but this is not set in stone, as the operator does control some resources directly and could conceivably also do so in this case, perhaps in the future.